src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/WebTransportTests.cs

```diff
@@ -0,0 +1,129 @@
+        var response1 = await controlStream2.ExpectSettingsAsync();
+
+        await Http3Api.ServerReceivedSettingsReader.ReadAsync().DefaultTimeout();
+
+        Assert.Equal(1, response1[(long)Http3SettingType.EnableWebTransport]);
+
+        var requestStream = await Http3Api.CreateRequestStream();
+        var headersConnectFrame = new[]
+        {
+            new KeyValuePair<string, string>(HeaderNames.Method, "CONNECT"),
+            new KeyValuePair<string, string>(HeaderNames.Protocol, "webtransport"),
+            new KeyValuePair<string, string>(HeaderNames.Scheme, "http"),
+            new KeyValuePair<string, string>(HeaderNames.Path, "/"),
+            new KeyValuePair<string, string>(HeaderNames.Authority, "server.example.com"),
+            new KeyValuePair<string, string>(HeaderNames.Origin, "server.example.com")
+        };
+
+        await requestStream.SendHeadersAsync(headersConnectFrame);
+        var response2 = await requestStream.ExpectHeadersAsync();
+
+        Assert.Equal((int)HttpStatusCode.OK, Convert.ToInt32(response2[HeaderNames.Status], null));
+
+        await requestStream.OnDisposedTask.DefaultTimeout();
+    }
+
+    [Theory]
+    [InlineData(
+        ((long)Http3ErrorCode.ProtocolError),
+        nameof(HeaderNames.Method), "GET", // incorrect method (verifies that webtransport doesn't break regular Http/3 get)
+        nameof(HeaderNames.Protocol), "webtransport",
+        nameof(HeaderNames.Scheme), "http",
+        nameof(HeaderNames.Path), "/",
+        nameof(HeaderNames.Authority), "server.example.com",
+        nameof(HeaderNames.Origin), "server.example.com")]
+    //[InlineData(
+    //    ((long)Http3ErrorCode..ProtocolError,
+    //    nameof(HeaderNames.Method), "CONNECT",
+    //    nameof(HeaderNames.Protocol), "webtransport",
+    //    nameof(HeaderNames.Scheme), "http", // incorrect scheme <-- it should be https but the tests don't work with it so I commented it out (also it is given by http/3)
+    //    nameof(HeaderNames.Path), "/",
+    //    nameof(HeaderNames.Authority), "server.example.com",
+    //    nameof(HeaderNames.Origin), "server.example.com")]
+    [InlineData(
+        ((long)Http3ErrorCode.ProtocolError),
+        nameof(HeaderNames.Method), "CONNECT",
+        nameof(HeaderNames.Protocol), "webtransport",
+        nameof(HeaderNames.Scheme), "http",
+        nameof(HeaderNames.Authority), "server.example.com",
+        nameof(HeaderNames.Origin), "server.example.com")]  // no path
+    [InlineData(
+        ((long)Http3ErrorCode.ProtocolError),
+        nameof(HeaderNames.Method), "CONNECT",
+        nameof(HeaderNames.Protocol), "webtransport",
+        nameof(HeaderNames.Scheme), "http",
+        nameof(HeaderNames.Path), "/",
+        nameof(HeaderNames.Origin), "server.example.com")]  // no authority
+    public async Task WebTransportHandshake_IncorrectHeadersRejects(long error, params string[] headers)
+    {
```

Comment:
```suggestion
    {
        _serviceContext.ServerOptions.AllowAlternateSchemes = true;

```

This should allow you to uncomment the "https" test case.

Reply:
Ok that is an interesting idea. However, as HTTP/3 is always over https, is there a need to even have this test?

Reply:
It might not be possible today in real scenarios, but we should still have and test logic to reject the "https" scheme. It's hard to say that there will never be a scenario in the future for the "http" scheme with HTTP/3 requests. The draft IETF WebTransport over HTTP/3 spec evidently felt it worth calling out even though HTTP/3 is always over https.

And evidently this *is* already possible in our tests, and I suspect it could be possible to get Kestrel to accept specially-crafted request with the "http" scheme (still over TLS) if the app developer set the public `KestrelServerOptions.AllowAlternateSchemes` to true themselves.